lazy_tensor_core/lazy_tensor_core/csrc/compiler/backend_impl_interface.h

```diff
@@ -1,21 +1,51 @@
 #pragma once
 
+#include <ATen/Tensor.h>
+
 #include <atomic>
 
-#include "lazy_tensor_core/csrc/compiler/node_lowering.h"
+#include "lazy_tensor_core/csrc/compiler/data.h"
 #include "lazy_tensor_core/csrc/lowering_context.h"
-#include "lazy_tensors/computation_client/computation_client.h"
+#include "lazy_tensors/shape.h"
 
 namespace torch_lazy_tensors {
 namespace compiler {
 
 class BackendImplInterface {
  public:
-  virtual std::unique_ptr<NodeLowering> CreateNodeLowering(
-      ir::LoweringContext* loctx) const = 0;
+  /**
+   * Initialization/Teardown
+   * */
+  // No-op by default. Allows custom functionality to be exposed through
+  // extension bindings.
+  virtual void InitializeAtenBindings() const {}
+
+  virtual void PrepareToExit() const = 0;
+
+  /**
+   * Configuration
+   * */
+
+  virtual void SetRngSeed(size_t seed) const = 0;
+
+  /**
+   * Data Transfer
+   * */
 
-  // For inference only.
-  virtual NodeLowering* GetNodeLowering() const = 0;
+  virtual BackendDataPtr MakeComputationDataFromTensor(
+      const at::Tensor& tensor, const lazy_tensors::Shape& shape,
+      const std::string& device) const = 0;
+
+  virtual BackendDataPtr CreateDataPlaceholder(std::string device,
+                                        lazy_tensors::Shape shape) const = 0;
```

Comment:
I'm not sure what the right source of truth is.  

From shape+dtype you can compute the buffer size, but if you are allocating buffers across sharded memories you may need more information to determine the per-shard size.

Given that, who makes the choice of optimal layout?  At the time we call CreateDataPlaceholder, have we completed XLA lowering/compilation yet?  I would imagine that during lowering, you could determine to use different layout for your weights than ATen default layout. 

If we pass you the stride of the AT tensor in this API, is it OK if it's always the default stride? I imagine that is what it would be, unless users are adjusting the stride of their tensor before making it an XLA tensor, as a workaround way to affect XLA performance.  IF default stride is OK, maybe you can just infer it;  if not, i'm not sure what the right user-API to choose a non-default stride would be?